docs/designs/2717-produce-tce-specific-releases.md

```diff
@@ -0,0 +1,223 @@
+# Produce community-edition releases that are decoupled from Tanzu Kubernetes Grid
+
+* Proposal:
+  [https://github.com/vmware-tanzu/community-edition/issues/2717](https://github.com/vmware-tanzu/community-edition/issues/2717)
+
+## Abstract
+
+At the time of writing, Tanzu Community Edition (TCE) relies on the Bill of
+Materials (BOMs) produced for VMware's paid product, Tanzu Kubernetes Grid
+(TKG). This means that creation of clusters, both management and workload,
+**always** use the same packages and versions as the rest of TKG. This binding
+to a TKG BOM was a decision made to ensure we could release TCE. Now we are at
+stage where we must unbind from TKG's BOM and begin producing TCE-specific BOMs.
+
+### Why
+
+Doing the above enables:
+
+* Building and packaging in the open, rather than behind VMware's network.
+* Packages that can be re-produced by users.
+* Releasing Kubernetes versions ahead of TKG.
+* Releasing core packages (e.g. antrea) ahead of TKG.
+* Releasing core-packages specific to TCE.
+* Patching TCE releases out-of-band from TKG.
+
+## Proposal
+
+### Creating a Core Package Repository
```

Comment:
I'm not seeing this called out, but the current BOM implementation uses a single registry URL, and we need to be able to support multiple repositories.

I think this aligns well with having a core repository that we control, and tools for users to set up and use repositories of their own.